md2/examples/md2sum.rs

```diff
@@ -0,0 +1,49 @@
+extern crate md2;
+
+use md2::{Md2, Digest};
+use std::env;
+use std::fs;
+use std::io::{self, Read};
+
+const BUFFER_SIZE: usize = 1024;
+
+/// Print digest result as hex string and name pair
+fn print_result(sum: &[u8], name: &str) {
+    for byte in sum {
+        print!("{:02x}", byte);
+    }
+    println!("\t{}", name);
+}
+
+/// Compute digest value for given `Reader` and print it
+/// On any error simply return without doing anything
+fn process<D: Digest + Default, R: Read>(reader: &mut R, name: &str) {
+    let mut sh: D = Default::default();
+    let mut buffer = [0u8; BUFFER_SIZE];
+    loop {
+        let n = match reader.read(&mut buffer) {
+            Ok(n) => n,
+            Err(_) => return,
+        };
+        sh.input(&buffer[..n]);
+        if n == 0 || n < BUFFER_SIZE {
+            break;
+        }
+    }
+    print_result(&sh.result(), name);
```

Comment:
[Now](https://github.com/fizyk20/generic-array/blob/master/src/hex.rs) we can write simply `println!("{:x}\t{}", sh.result(), name)`, so `print_result` function is not necessary.